packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3099,10 +3100,19 @@ export class TransactionController extends BaseController<
           const updatedTransactionMeta = this.#getTransaction(
             transactionId,
           ) as TransactionMeta;
-          this.messenger.publish(`${controllerName}:transactionApproved`, {
-            transactionMeta: updatedTransactionMeta,
-            actionId,
-          });
+
+          if (approvalResult === ApprovalState.NotApproved) {
+            this.messenger.publish(`${controllerName}:transactionFailed`, {
```

Comment:
This is already published in `#failTransaction` whenever the status is changed to failed.

So we can just check `ApprovalState.Approved` here instead, before publishing approved event.